lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1122,18 +1122,18 @@ export class UserAgentApplication {
     const account: Account = accountObj || this.getAccount();
 
     // This is a final check for all queryParams added so far; preference order: sid > login_hint
-    // sid cannot be passed along with login_hint, hence we check both are not populated yet in queryParameters so far
-    if (account) {
+    // sid cannot be passed along with login_hint or domain_hint, hence we check both are not populated yet in queryParameters so far
+    if (account && !qParams[SSOTypes.SID]) {
       // sid
       if (account.sid && serverReqParams.promptValue === PromptState.NONE) {
-        if (!qParams[SSOTypes.SID]  && !qParams[SSOTypes.LOGIN_HINT]) {
+        if (!qParams[SSOTypes.LOGIN_HINT]) {
```

Comment:
You can simplify this by moving the '!qParams[SSOTypes.LOGIN_HINT]' check to line 1126.

Reply:
+1